torch/_dynamo/variables/builder.py

```diff
@@ -570,14 +571,20 @@ def build_key_value(k, v):
         elif isinstance(value, HigherOrderOperator):
             self.install_guards(GuardBuilder.TYPE_MATCH, GuardBuilder.NAME_MATCH)
             return TorchHigherOrderOperatorVariable.make(value, source=self.source)
+        elif isinstance(value, torch.cuda.StreamContext):
+            self.install_guards(GuardBuilder.ID_MATCH)
+            stream_source = AttrSource(self.source, "stream")
+            stream_var = VariableBuilder(self.tx, stream_source)(value.stream)
+            return StreamContextVariable.create_from_stream(self.tx, stream_var)
         elif isinstance(value, _StreamBase):
             self.install_guards(GuardBuilder.ID_MATCH)
-            return StreamVariable(
+            stream_var = StreamVariable(
                 None,
                 value,
-                value.device.type,
+                value.device,
                 source=self.source,
             )
+            return self.tx.output.side_effects.track_object_existing(value, stream_var)
```

Comment:
Maybe don't need this yet.

Reply:
Do we support mutations on streams? I thought they're pretty immutable.